### PR TITLE
WIP: Extend functionalities of the Upload component

### PR DESCRIPTION
**Pull request by Dhaiwat10**
## What kind of change does this PR introduce?
Extended the present functionality of the `Upload` component. (Issue #254)

## What is the current behavior?
The `Upload' component didn't support 'tracking'/retrieval of the uploaded files.

## What is the new behavior?
• Added two props: files and setFiles which allow you to control the uploaded files using state from a parent component.

• Added previews for uploaded images in the story.

## Additional context
image

### CODE REVIEW

**Comment by Dhaiwat10** [comment]
Planning to add the ability to delete any of the uploaded files. Let me know what you think.

File: src/components/Upload/Upload.tsx [hunk 1]
```diff
@@ -54,10 +62,20 @@ function Dragger({ label, afterLabel, beforeLabel, layout, children }: any) {
...
-    const files = e.dataTransfer.files
-    console.log(files)
+    console.log('fileDrop')
+    const newFiles = e.dataTransfer.files
+    console.log(newFiles)
     console.log(classes)
+    setFiles([...files, ...newFiles])
+  }
+
+  const fileUpload = (e: any) => {
```
Comment (MildTomato) [inline src/components/Upload/Upload.tsx:72]: can you update the types to not use `any` ?

File: src/components/Upload/Upload.tsx [hunk 2]
```diff
@@ -54,10 +62,20 @@ function Dragger({ label, afterLabel, beforeLabel, layout, children }: any) {
...
+    e.preventDefault()
+    // console.log('fileUpload')
+    const newFiles = e.target.files
+    console.log(newFiles)
+    console.log(classes)
```
Comment (MildTomato) [inline src/components/Upload/Upload.tsx:77]: remove console logs

**Review by MildTomato — CHANGES_REQUESTED** [review]
looks good, just a couple of comments

**Comment by Dhaiwat10** [comment]
Done ✅

File: src/components/Upload/Upload.tsx [hunk 3]
```diff
@@ -54,10 +62,18 @@ function Dragger({ label, afterLabel, beforeLabel, layout, children }: any) {
 
   const fileDrop = (e: any) => {
     e.preventDefault()
-    // console.log('fileDrop')
-    const files = e.dataTransfer.files
-    console.log(files)
+    console.log('fileDrop')
+    const newFiles = e.dataTransfer.files
+    console.log(newFiles)
```
Comment (MildTomato) [inline src/components/Upload/Upload.tsx:67]: remove console logs 😄

**Review by MildTomato — CHANGES_REQUESTED** [review]
just some stray console logs

**Comment by Dhaiwat10** [comment]
classic commit 🙃

**Review by MildTomato — APPROVED** [review]

**Pull request merged**